In [15]:
import plotly.express as px
import pandas as pd

df = pd.read_csv('data2.csv')

df

學年度別       大專院校       高級中學       職業學校       國民中學       國民小學        幼兒園
0    70年   53049.96   21616.75   28353.78   20027.09   12254.20          -
1    71年   55869.18   21656.14   43475.99   23168.90   14872.66          -
2    72年   65000.68   22053.02   43949.43   22638.55   14346.20          -
3    73年   60808.55   15243.66   39050.71   25361.06   14743.58          -
4    74年   65651.22   26571.33   40051.93   27695.85   16572.49          -
5    75年  155901.90   27951.36   73079.23   28979.22   16476.64          -
6    76年  134802.27   29217.68   59202.94   33613.08   21366.48          -
7    77年  118537.37   34858.81   48890.74   35517.77   22985.32          -
8    78年  118475.42   45829.51   62344.37   44340.98   26422.17          -
9    79年   89860.38   65096.11   78563.30   52442.72   35821.83          -
10   80年  157250.56   68831.46   78377.21   60725.78   39698.97          -
11   81年  155115.27   77965.66   84089.63   69638.33   48621.09          -
12   82年  118408.60   69844.22   80615.78   67409.85   48462.60          -
13   83年  130255.32   77056.06   82483.01   68970.20   51869.02          -
14   84年  176787.61  101799.55  111455.45   92162.85   68533.42          -
15   85年  208012.96  102416.10  123776.42  100352.03   75284.19          -
16   86年  256788.35  101891.89  134775.43  107734.20   82192.99          -
17   87年  245605.87  103663.34  139958.76  115033.64   83906.20          -
18   88年  216732.81   93041.52  134328.21  106853.25   74877.73          -
19   89年  196563.35   94937.30  135874.14  109939.82   80602.14     110682
20   90年  218868.84  105908.52  144164.52  117757.92   94168.19  113588.46
21   91年  241130.10  115054.77  151157.52  126319.01  104359.83   110538.7
22   92年  246120.23  118911.65  152201.71  128385.84  100066.35  116524.71
23   93年  253497.67  119211.51  150800.88  131288.74  102955.84  118127.62
24   94年  281463.24  121625.28  149657.50  135723.02  109495.80  125348.64
25   95年  338923.00  128703.46  153282.67  139398.42  113158.47  129555.86
26   96年  431273.36  136395.11  154026.50  139615.45  116646.16  132922.52
27   97年  376055.56  137790.31  156891.62  139656.97  125590.13  131276.07
28   98年  228893.57  138340.27  162634.60  143532.48  135681.02   131483.8
29   99年  209009.04  143672.45  171986.09  153784.37  143879.68  137893.22
30  100年  279114.25  144867.70  173129.16  166004.86  152426.15  146834.17
31  101年  272399.15  143357.72  172604.68  176348.22  164988.78  142445.84
32  102年  225141.80  143436.20  182407.31  185405.85  174236.47  154100.94
33  103年  229436.89  150909.58  190253.09  193500.26  173009.18   150694.1
34  104年  231505.17  158155.40  198676.41  205477.51  172581.86  151717.91
35  105年  246171.11  166939.94  208719.47  222809.09  178453.12  144961.39
36  106年  247470.65  175004.92  214456.06  233446.56  183476.03  144156.84
37  107年  236924.26  180470.10  226628.91  249781.33  183585.81  164649.62
38  108年  232504.61  176064.45  232063.41  257286.85  184035.16  159239.47
39  109年  212458.24  176679.70  232030.62  252483.85  180734.08  136983.48
40  110年  200339.72  179964.32  221316.23  252617.82  177096.89  143346.53

In [17]:
column_names = ['大專院校', '高級中學', '職業學校', '國民中學','國民小學']
fig = px.line(df, x="學年度別", y=column_names,
              hover_data={"學年度別", "學年度別"},
              title='70-110各級學校每生擁有經費')
fig.update_xaxes(
    dtick="M1",
    tickformat="%b\n%Y",
    ticklabelmode="period")
fig.show()

#### Q1:請問最新的年度中各級學校每生擁有經費依序為何?
#### A: 國中 => 職校 => 大專院校 => 高中 => 國小

#### Q2:圖中可以看到原本大專院校一直都是經費最多的，但後續被國中及職校超越，請問分別是在幾年?可能原因為何?
#### A: 國中在106~107年度間超越大專院校，職校於108~109年度間超越大專院校，推測政府因推動108課綱而增加經費

#### Q3:大專院校在哪年擁有歷史最高經費?可能原因為何?
#### A: 96年度，推測因開始推動大學多元入學方案，而提高大專院校經費